### ISSUE TITLE
CA1508 fires for Debug.Assert

### ISSUE THREAD

**Issue by tannergooding** [dotnet/roslyn-analyzers#6983]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Steps To Reproduce

1. Create a new C# Class Library, targeting `net8.0`
2. Edit the csproj and add `<AnalysisLevel>latest-all</AnalysisLevel>`
3. Define the following class:
```csharp
using System.Diagnostics;

namespace MyNamespace;

public static class MyClass1
{
    internal const int MyConstant = 16;

    public static void MyMethod()
    {
        Debug.Assert(MyConstant == 16);
    }
}
```

### Expected behavior

The analyzer should not trigger for a `Debug.Assert`. This is often used to handle code that may make assumptions based on the value of a constant, such as not handling larger code paths or having pre-defined static data and where it may need to be updated if the value of the constant ever changes.

### Actual behavior

The analyzer is triggered.
